### PR TITLE
Migrations

### PR DESCRIPTION
**Pull request by dylan-harper**
# Description

Please check the type of change your PR introduces:
- [ ] Bugfix (change which fixes an issue)
- [ ] Feature (change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [x] Documentation content changes
- [x] Other (please describe): User and Story Migrations

 Migrated users and stories table. Created models, factories, validations and related specs. --> 

Dependencies added:
FactoryBot, Faker

### Type of change

Please check the relevant option.

- [x] Non-Breaking Change
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Big Breaking change (this breaks everything and I am sorry)

# Where should reviewers start?

Start with taking a look at the schema and the data types I chose, those are up for discussion as always. Everything else is outlined and pretty straightforward, not a big/complicated PR. 

# How Has This Been Tested?

Please check the option(s) that best describe your tests. 

- [x] New spec files were added
- [ ] Tests were added to existing spec files
- [ ] Some tests were changed
- [ ] No tests were changed
- [ ] All tests were changed (please explain what in God's name happened)

### Tests Description

- Validations for User and Story. 

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have removed all unused code or commented why I left it
- [ ] I have deployed my changes to a test-heroku and they work
- [x] My changes generate no new warnings
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: spec/factories/story.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :story do
+    user_id { Faker::Number.number(digits: 2) }
+    total_time { Faker::Number.number(digits: 3) }
+    body_text { Faker::Lorem.paragraph(sentence_count: 4) }
+    image { Faker::Lorem.sentence }
```
Comment (jbreit88) [inline spec/factories/story.rb]: I think there is a [Faker generator](https://rubydoc.info/github/faker-ruby/faker/master/Faker%2FInternet.url) for URLs. I anticipate that we will be passing the images and maybe the sounds as URLs, so if we could get that data looking as close to the end product as possible I think that would be good.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class User < ApplicationRecord
+  validates_presence_of :username, :email
+
```
Comment (kevingloss) [inline app/models/user.rb:6]: If we want to add `gem 'email_validator` to the gemfile we can add the following in here to check emails: `validates :email, email: true, uniqueness: true`

File: db/schema.rb [hunk 3]
```diff
@@ -10,9 +10,27 @@
...
+    t.datetime "updated_at", null: false
+  end
+
+  create_table "users", force: :cascade do |t|
+    t.string "username"
+    t.string "email"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
```
Comment (kevingloss) [inline db/schema.rb:36]: Looks like we need to add a reference to tie the user and stories together.

**Review by kevingloss — CHANGES_REQUESTED** [review]
Couple of small comments to incorporate. Overall it looks great. Thanks Dylan.

File: app/models/user.rb (hunk 2 above)
Comment (dylan-harper) [reply app/models/user.rb:6]: I added the email gem and a line to test for uniqueness of email in the user model

File: db/schema.rb (hunk 3 above)
Comment (dylan-harper) [reply db/schema.rb:36]: Good catch! I added the foreign key to the schema

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  has_many :stories
+
```
Comment (JCNapier) [inline app/models/user.rb:3]: I am not sure if any of you have used this in the past, (I know a past group of mine has) but do think we should include a dependent destroy here for the user model? That way if the user is destroyed, the stories associated with that user are also destroyed?

**Review by JCNapier — APPROVED** [review]
Everything looks really solid here. Thank you for doing this Dylan.

**Review by JCNapier — COMMENTED** [review]
I like the greek philosopher names being used in the Faker haha

File: app/models/user.rb (hunk 4 above)
Comment (dylan-harper) [reply app/models/user.rb:3]: 100% Super good idea. I'll add it

**Review by JCNapier — APPROVED** [review]
Awesome, looks Great man. You Rock.

**Review by jbreit88 — COMMENTED** [review]
Hey this looks great! Just had one comment about the Faker data. Change or not though, either is fine.

File: spec/factories/story.rb (hunk 1 above)
Comment (dylan-harper) [reply spec/factories/story.rb]: Good find! Image and sound are now fake urls instead of just strings

**Review by kevingloss — APPROVED** [review]

**Pull request merged**